common/random.cpp

```diff
@@ -73,5 +72,13 @@ uint RandomSource::getRandomNumberRng(uint min, uint max) {
 int RandomSource::getRandomNumberRngSigned(int min, int max) {
 	return getRandomNumber(max - min) + min;
 }
+		
+inline void RandomSource::scrambleSeed() {
+	//marsaglia's paper says that any of 81 triplets are feasible
+	//(11,21,13) was chosen, with (cba) and (>>,<<,>>)
```

Comment:
Reading through the paper, Marsaglia suggests a specific triplet for 32-bit seeds, which will pass almost all sets of randomness: << 13 , >> 17, << 5. Further down he does say that any of the choices he provides in the paper will do as well, but he has extensively tested specific ones like this. Perhaps we should use that triplet?

Reply:
The exact wording in the paper is "It uses one of my favorite choices, [a, b, c] = [13, 17, 5], and will pass almost all tests of randomness, except the binary rank test in Diehard." For context, Diehard is an older battery of tests developed to test PRNGs, and has been superseded by more modern tests, such as Dieharder and TestU01. 

Using TestU01's SmallCrush on the Xorshift* version of this algorithm, all triplets and variations pass the exact same tests and fail BirthdaySpacings, a test that would be impossible to feel in a non-cryptographic context. So, personal preference notwithstanding, there's no real benefit to choosing certain triplets above others for this.

Marsaglia's comment above on the matter is that (13,17,5) is one of his personal favorites, and in much the same way, (11,21,13) is a personal favorite of mine :)

Reply:
I decided to throw TestU01 at all possible xorshift* pairings, with initial seed = 1, and the best result for BirthdaySpacings still fails, with p = 1.6*10^-16 (>>2, <<7, >>9). This is better than Marsaglia or my preferences, but still fails the test overall. I would not mind using this set of values instead, but it certainly would feel like splitting hairs.